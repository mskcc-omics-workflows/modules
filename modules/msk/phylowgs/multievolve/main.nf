process PHYLOWGS_MULTIEVOLVE {
    tag "$meta.id"
    label 'process_high'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/phylowgs:v1.5-msk':
        'docker.io/mskcc/phylowgs:v1.5-msk' }"

    input:
    tuple val(meta), path(cnv_data), path(ssm_data)

    output:
    tuple val(meta), path("chains/trees.zip")   , emit: trees
    path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def threads = task.cpus * 2

    """
    python2 \\
        /usr/bin/multievolve.py  \\
        ${args} \\
        --num-chains ${threads} \\
        --ssms ${ssm_data} \\
        --cnvs ${cnv_data}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phylowgs: \$PHYLOWGS_TAG
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir chains
    touch chains/trees.zip

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phylowgs: \$PHYLOWGS_TAG
    END_VERSIONS
    """
}
