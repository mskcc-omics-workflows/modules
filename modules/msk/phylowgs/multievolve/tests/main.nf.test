nextflow_process {

    name "Test Process PHYLOWGS_MULTIEVOLVE"
    script "../main.nf"
    process "PHYLOWGS_MULTIEVOLVE"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "phylowgs"
    tag "phylowgs/multievolve"

    test("phylowgs_multievolve - zip") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file(params.test_data_mskcc['neoantigen']['cnv_data'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['ssm_data'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.trees[0][1]).name).match() }

            )
        }

    }

    test("phylowgs_multievolve - zip - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file('cnv_data'),
                    file('ssm_data')
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }


}
