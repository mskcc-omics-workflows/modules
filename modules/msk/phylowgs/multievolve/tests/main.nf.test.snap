{
    "phylowgs_multievolve - zip": {
        "content": [
            "trees.zip"
        ],
        "timestamp": "2024-06-11T18:01:56.174868"
    },
    "phylowgs_multievolve - zip - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "trees.zip:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2eaafdf0e85898fdc2e5d40dbbfee2f7"
                ],
                "trees": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "trees.zip:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2eaafdf0e85898fdc2e5d40dbbfee2f7"
                ]
            }
        ],
        "timestamp": "2024-06-11T18:02:03.678335"
    }
}